tests/api/documents/test_documents_add_cover_url_extension.py

```
@@ -0,0 +1,248 @@
+
+    # Mock get_thumbnail_url
+    with mock.patch("rero_ils.modules.documents.extensions.add_cover_url.get_thumbnail_url") as mock_get_thumbnail:
+        mock_get_thumbnail.return_value = "https://example.com/cover.jpg", "TestProvider"
+
+        extension.add_cover_url(document)
+
+        # Verify get_thumbnail_url was called with the smallest ISBN first
+        mock_get_thumbnail.assert_called_once_with("9780000000000", cached=True)
+
+
+def test_add_cover_url_extension_exception_then_success(app):
```

Ruff (0.15.2)

[warning] Unused function argument: `app`

(ARG001)